heudiconv/convert.py

```diff
@@ -922,6 +923,42 @@ def save_converted_files(
         # we should provide specific handling for fmap,
         # dwi etc which might spit out multiple files
 
+        # Also copy BIDS files although they might need to
+        # be merged/postprocessed later
+        bids_files = (
+            sorted(bids_files)
+            if len(bids_files) == len(res_files)
+            else [None] * len(res_files)
+        )
+        # preload since will be used in multiple spots
+        bids_metas = [load_json(b) for b in bids_files if b]
+
+        # interrupted scans on XA: see dcm2niix #742
```

Comment:
this code block is worth moving into some helper function e.g. `handle_partial_volumes(bids_files, bids_metas)` which would potentially mutate those structures.